Exception thrown in `add-appt` command when doctor/patient name is not in the displayed list

When the `add-appt` is has a doctor/patient name that does not exist in the list currently displayed, an exception will be thrown. This makes it impossible to enter the command (ie. pressing the enter key will not work).

Instead, an error message explaining that the doctor/patient name does not exist should be displayed.

Steps to reproduce:

```
add-doctor n/John Doe p/98765432 e/johndoe@example.com a/123 Clementi Ave 3 s/Orthopedics
add-patient n/Jane Doe p/91234567 e/janedoe@example.com a/456 Clementi Ave 3 d/01-05-1990 g/F t/friend
add-appt dn/Jane Do pn/John Doe d/23-04-1987 t/1100
```

Part of the exception thrown:

```
Caused by: java.util.NoSuchElementException: No value present
        at java.base/java.util.Optional.get(Optional.java:143)
        at seedu.address.logic.commands.AddAppointmentCommand.execute(AddAppointmentCommand.java:79)
        at seedu.address.logic.LogicManager.execute(LogicManager.java:51)
        at seedu.address.ui.MainWindow.executeCommand(MainWindow.java:177)
        at seedu.address.ui.CommandBox.handleCommandEntered(CommandBox.java:45)
```

Most likely has to do with directly calling `Optional.get()` in [AddAppointmentCommand.java#L79-L80](https://github.com/AY2425S1-CS2103T-T09-3/tp/blob/e092a8d500cd8b989b395c56e92362480c1e8c5b/src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java#L79-L80), we should handle the case where the doctor/patient name does not exist and `Optional` is empty.

--- a/src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java
+++ b/src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java
@@ -46,6 +46,8 @@ public class AddAppointmentCommand extends Command {
 
     public static final String MESSAGE_SUCCESS = "New appointment added: %1$s";
     public static final String MESSAGE_DUPLICATE_APPOINTMENT = "This appointment already exists in the address book";
+    public static final String MESSAGE_INVALID_PATIENT_DISPLAYED_NAME = "Unable to find patient with name: ";
+    public static final String MESSAGE_INVALID_DOCTOR_DISPLAYED_NAME = "Unable to find doctor with name: ";
 
     private final Name patientName;
     private final Name doctorName;
@@ -76,8 +78,12 @@ public class AddAppointmentCommand extends Command {
                 .filter(person -> person instanceof Doctor && person.getName().equals(doctorName))
                 .findAny();
 
-        Person patientToEdit = patient.get();
-        Person doctorToEdit = doctor.get();
+        Person doctorToEdit = doctor.orElseThrow(
+                () -> new CommandException(MESSAGE_INVALID_DOCTOR_DISPLAYED_NAME + doctorName));
+
+        Person patientToEdit = patient.orElseThrow(
+                () -> new CommandException(MESSAGE_INVALID_PATIENT_DISPLAYED_NAME + patientName));
+
 
         Person editedPatient;
         Person editedDoctor;

--- a/src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java
+++ b/src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java
@@ -78,11 +78,11 @@ public class AddAppointmentCommand extends Command {
                 .filter(person -> person instanceof Doctor && person.getName().equals(doctorName))
                 .findAny();
 
-        Person doctorToEdit = doctor.orElseThrow(
-                () -> new CommandException(MESSAGE_INVALID_DOCTOR_DISPLAYED_NAME + doctorName));
+        Person doctorToEdit = doctor.orElseThrow(() ->
+                new CommandException(MESSAGE_INVALID_DOCTOR_DISPLAYED_NAME + doctorName));
 
-        Person patientToEdit = patient.orElseThrow(
-                () -> new CommandException(MESSAGE_INVALID_PATIENT_DISPLAYED_NAME + patientName));
+        Person patientToEdit = patient.orElseThrow(() ->
+                new CommandException(MESSAGE_INVALID_PATIENT_DISPLAYED_NAME + patientName));
 
 
         Person editedPatient;

